lib/backend/k8s/k8s.go

```diff
@@ -662,6 +671,40 @@ func (c *KubeClient) getPolicy(k model.PolicyKey) (*model.KVPair, error) {
 	}
 }
 
+// deletePolicy deletes a Policy. It takes care of appropriate conversions whether
+// the Policy is backed by a NetworkPolicy or a Global Network Policy.
+func (c *KubeClient) deletePolicy(d *model.KVPair) error {
+	k, ok := d.Key.(model.PolicyKey)
+	if !ok {
+		return goerrors.New("Incorrect key")
+	} else if k.Name == "" {
+		return goerrors.New("Missing policy name")
+	}
+	if strings.HasPrefix(k.Name, "knp.default.") {
```

Comment:
AFAIK we don't need to implement delete for knp.default policies, unless I'm forgetting something.

These should pretty much be exclusively managed through the k8s API, so we should probably just return an "operation not supported" error.

Ultimately I think `calicoctl` should prevent people from deleting / modifying `knp.default` policies anyway for both etcd and kdd.

Discussed with @lxpollitt and he agrees.